src/client/mod.rs

```diff
@@ -107,18 +110,28 @@ impl Client {
         Client::with_connector(Pool::new(config))
     }
 
+    pub fn with_proxy_config(config: proxy::Config) -> Client {
```

Comment:
What's the difference between using a Connector with_proxy, and calling `client.add_proxy`?